cmd/ocsp-responder/main.go

```diff
@@ -316,7 +478,36 @@ as generated by Boulder's ceremony command.
 		filter, err := newFilter(issuerCerts, c.OCSPResponder.RequiredSerialPrefixes)
 		cmd.FailOnError(err, "Couldn't create OCSP filter")
 
-		source = &dbSource{dbMap, filter, c.OCSPResponder.Timeout.Duration, logger}
+		pLookup := dbReceiver{dbMap, filter, logger}
+
+		// Set up the redis source if there is a config. Otherwise just
+		// set up a mysql source.
+		if c.OCSPResponder.Redis.Addrs != nil {
+			logger.Info("redis config found, configuring redis reader")
+			rocspReader, err := rocsp_config.MakeReadClient(&c.OCSPResponder.Redis, clk)
+			if err != nil {
+				cmd.FailOnError(err, "could not make redis client")
+			}
+			source = &dbSource{
+				primaryLookup:   pLookup,
+				secondaryLookup: redisReceiver{rocspReader},
+				filter:          filter,
+				timeout:         c.OCSPResponder.Timeout.Duration,
+				log:             logger,
+				metrics:         stats,
+			}
+		} else {
+			logger.Info("no redis config found, using mysql as only ocsp source")
+			source = &dbSource{
+				primaryLookup:   pLookup,
+				secondaryLookup: nil,
+				filter:          filter,
+				timeout:         c.OCSPResponder.Timeout.Duration,
+				log:             logger,
+				metrics:         stats,
+			}
```

Comment:
You can reduce duplication here like so:

```
var redisLookup ocspLookup
if c.OCSPResponder.Redis.Addrs != nil {
   ....
  redisLookup = redisReceiver{rocspReader}
```

Reply:
Thanks. That should have been obvious to me when I was trying to think of how to deduplicate that code. Done.